presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayFunctions.java

```diff
@@ -54,6 +63,25 @@ public static Boolean contains(@SqlType("array<double>") Slice slice, @SqlType(S
     @SqlType(StandardTypes.BOOLEAN)
     public static Boolean contains(@SqlType("array<varchar>") Slice slice, @SqlType(StandardTypes.VARCHAR) Slice value)
     {
-        return JsonFunctions.jsonArrayContains(slice, value);
+        return arrayContains(slice, VarcharType.VARCHAR, value);
+    }
+
+    private static Boolean arrayContains(Slice slice, Type type, Object value)
+    {
+        Block block = readBlock(slice.getInput());
+        Block valueBlock = TypeUtils.createBlock(type, value);
+
+        for (int i = 0; i < block.getPositionCount(); i++) {
+            if (type.equalTo(block, i, valueBlock, 0)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static Long arrayLength(Slice slice)
```

Comment:
Since this is only used by the cardinality aggregation move it to ArrayCardinalityAggregation